components/org.wso2.carbon.identity.oauth.scope.endpoint/src/main/java/org/wso2/carbon/identity/oauth/scope/endpoint/exceptionMapper/JsonProcessingExceptionMapper.java

```diff
@@ -0,0 +1,58 @@
+
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.scope.endpoint.exceptionMapper;
+
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.oauth.scope.endpoint.dto.ErrorDTO;
+
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+/**
+ * Handles exceptions when an incorrect json requests body is received.
+ * Sends a default error response.
+ */
+public class JsonProcessingExceptionMapper implements ExceptionMapper<UnrecognizedPropertyException> {
+
+    private static final Log log = LogFactory.getLog(JsonProcessingExceptionMapper.class);
+    public static final String ERROR_MESSAGE = "Invalid Request";
+    public static final String DESCRIPTION = "Provided JSON request content is not in the valid format.";
+
+    @Override
+    public Response toResponse(UnrecognizedPropertyException e) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Provided JSON request content is not in the valid format:", e);
+        }
+
+        ErrorDTO errorDTO = new ErrorDTO();
+        errorDTO.setCode(Response.Status.BAD_REQUEST.toString());
+        errorDTO.setMessage(ERROR_MESSAGE);
+        errorDTO.setDescription(DESCRIPTION);
```

Comment:
Shall we put the actual description in the error message? That would be more meaningful